openpype/modules/deadline/abstract_submit_deadline.py

```diff
@@ -218,146 +309,38 @@ class DeadlineJobInfo(object):
+    ExtraInfoKeyValue = attr.ib(factory=partial(DeadlineKeyValueVar,
+                                                "ExtraInfoKeyValue"))
 
     # Task Extra Info Names
     # ----------------------------------------------
     OverrideTaskExtraInfoNames = attr.ib(default=None)  # Default: false
-    _taskExtraInfos = attr.ib(factory=list)
-
-    @property
-    def TaskExtraInfoName(self):  # noqa: N802
-        """Return all TaskExtraInfoName values formatted for Deadline.
-
-        Returns:
-            dict: as `{'TaskExtraInfoName0': 'value'}`
-
-        """
-        out = {}
-        for index, v in enumerate(self._taskExtraInfos):
-            out["TaskExtraInfoName{}".format(index)] = v
-        return out
-
-    @TaskExtraInfoName.setter
-    def TaskExtraInfoName(self, val):  # noqa: N802
-        self._taskExtraInfos.append(val)
+    TaskExtraInfoName = attr.ib(factory=partial(DeadlineIndexedVar,
+                                                "TaskExtraInfoName"))
 
     # Output
     # ----------------------------------------------
-    _outputFilename = attr.ib(factory=list)
-    _outputFilenameTile = attr.ib(factory=list)
-    _outputDirectory = attr.ib(factory=list)
-
-    @property
-    def OutputFilename(self):  # noqa: N802
-        """Return all OutputFilename values formatted for Deadline.
-
-        Returns:
-            dict: as `{'OutputFilename0': 'filename'}`
-
-        """
-        out = {}
-        for index, v in enumerate(self._outputFilename):
-            out["OutputFilename{}".format(index)] = v
-        return out
-
-    @OutputFilename.setter
-    def OutputFilename(self, val):  # noqa: N802
-        self._outputFilename.append(val)
-
-    @property
-    def OutputFilenameTile(self):  # noqa: N802
-        """Return all OutputFilename#Tile values formatted for Deadline.
-
-        Returns:
-            dict: as `{'OutputFilenme#Tile': 'tile'}`
-
-        """
-        out = {}
-        for index, v in enumerate(self._outputFilenameTile):
-            out["OutputFilename{}Tile".format(index)] = v
-        return out
-
-    @OutputFilenameTile.setter
-    def OutputFilenameTile(self, val):  # noqa: N802
-        self._outputFilenameTile.append(val)
-
-    @property
-    def OutputDirectory(self):  # noqa: N802
-        """Return all OutputDirectory values formatted for Deadline.
-
-        Returns:
-            dict: as `{'OutputDirectory0': 'dir'}`
-
-        """
-        out = {}
-        for index, v in enumerate(self._outputDirectory):
-            out["OutputDirectory{}".format(index)] = v
-        return out
-
-    @OutputDirectory.setter
-    def OutputDirectory(self, val):  # noqa: N802
-        self._outputDirectory.append(val)
+    OutputFilename = attr.ib(factory=partial(DeadlineIndexedVar,
+                                             "OutputFilename"))
+    OutputFilenameTile = attr.ib(factory=partial(DeadlineIndexedVar,
+                                                 "OutputFilename{}Tile"))
```

Comment:
Anyone know of a better more pythonic way to initialize these with `attr`? I feel like I'm missing something obvious.

Reply:
@kalisp or @antirotor do you know?